### ISSUE TITLE
unnested_or_patterns can actually expand code (in text size)

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6973]
````rust
#![warn(clippy::unnested_or_patterns)]
pub enum E {
    A,
    B,
}


pub fn a(enm: &E) {
    match enm {
        &E::A | &E::B => {} 
    }
}


pub fn main() {}
````

clippy will suggest to use
`&(E::A | E::B)` instead of
`&E::A | &E::B`
this actually makes the code a bit bigger.
Is this intended?

**Comment by matthiaskrgr** [comment]
````
57 |         Some(&b'e' | &b'E') => {
   |         ^^^^^^^^^^^^^^^^^^^
````

````
help: nest the patterns
   |
57 |         Some(&(b'e' | b'E')) => {
   |         ^^^^^^^^^^^^^^^^^^^^

warning: unnested or-patterns
````

:thinking:

**Comment by camsteffen** [comment]
Seems reasonable to ignore `&`.
